### PR TITLE
Fix #10: add search parameters and docs

### LINKED ISSUE #10: Add support for search API
Docs: https://github.com/rokka-io/rokka-site/blob/user-metadata/source/_references/searching.md#searching-for-particular-images

### PR DESCRIPTION
**Pull request by mweibel**

### CODE REVIEW

File: src/apis/sourceimages.js [hunk 1]
```diff
@@ -11,27 +11,62 @@ export default (state) => {
...
+   *
    * ```js
    * rokka.sourceimages.list('myorg')
    * 	 .then(function(result) {})
    * 	 .catch(function(err) {});
    * ```
    *
+   * Searching for images can be achieved using the `search` parameter.
+   * Supported are predefined fields like `height`, `name` etc. but also user metadata.
+   * If you search for user metadata, the field name has to be prefixed with `user:TYPE`.
```
Comment (pstadler) [inline src/apis/sourceimages.js]: Would be great to state how different queries work (i.e. search without one of these fields) and if it's possible to use `OR`, `AND`, etc.

File: README.md [hunk 2]
```diff
@@ -129,16 +129,40 @@ rokka.memberships.create('myorg', 'user@example.org', rokka.memberships.ROLES.WR
 
 ### Source Images
 
-#### rokka.sourceimages.list(organization, [limit=null], [offset=null]) → Promise
+#### rokka.sourceimages.list(organization, [limit=null], [offset=null], [sort=null], [search=null]) → Promise
```
Comment (pstadler) [inline README.md]: Please change this to take an object instead: `{ limit: <optional>, offset: <optional>, search: <optional> }`

File: src/apis/sourceimages.js (hunk 1 above)
Comment (pstadler) [inline src/apis/sourceimages.js]: I see that this is done below, sorry for that. Maybe still add a more complex query as example?

Comment (mweibel) [inline src/apis/sourceimages.js]: fixed

File: README.md (hunk 2 above)
Comment (mweibel) [inline README.md]: fixed

File: src/apis/sourceimages.js [hunk 3]
```diff
@@ -11,27 +11,63 @@ export default (state) => {
...
+   *
+   * Sorting works with user metadata as well and can be passed as either an array or as a
+   * comma separated string.
+   *
    * @authenticated
-   * @param  {string} organization  name
-   * @param  {number} [limit=null]
-   * @param  {number} [offset=null]
+   * @param  {string}       organization  name
+   * @param  {number}       [limit=null]
```
Comment (pstadler) [inline src/apis/sourceimages.js]: docs are wrong, this is an object now.

Comment (mweibel) [inline src/apis/sourceimages.js]: Nah, they're correct. Or at least the least incorrect. 
Tried various options, none I really like:
```
@param {Object} qp Query params
@param {number} [qp.limit=null] Limit
// etc. 
```
Problem with that: there's no qp param

```
@param {{limit: number, offset: number}} qp
```
Same issue: no qp, plus having all the possible things within one line is ugly.

Comment (pstadler) [inline src/apis/sourceimages.js]: But they're composed wrongly in the README, looks like this function takes positional arguments

**Pull request merged**